Fix ActiveStorage::Service::AzureService#url

ffmike opened the pull request:
Add in the content_type option, which is defined as part of the generic ActiveStorage::Service class.

Without this option, attempts to generate a URL for an attached file fail with "ArgumentError (unknown keyword: content_type)"

kirs committed:
Do not let use `serialize` on native JSON/array column

kamipo committed:
Use `Concurrent::Map` than `Mutex` and `Mutex_m` for statement caches

Statement caches are used as a concurrent map. It will more clarify to
using `Concurrent::Map`.

gladtocode committed:
Improve bundler binstub error message

Rails displays an error message if you have a bundler-generated binstub
 at `./bin/rails` instead of a Rails-generated binstub.

This error message is misleading because it makes it seem as though
Rails 5 introduced recent changes in how binstubs are used, when these
changes were actually introduced way back in Rails 4.

The suggested upgrade steps are appropriate for an app that was created
in Rails 3, but they likely aren't the correct fix for someone who sees
this error message today on a modern app.  I believe the `--binstubs`
option on bundler is a more likely culprit and troubleshooting path.

Commit:
Allow non-English values for collection_radio_buttons/check_boxes

y-yagi committed:
Remove extra empty line

ricardotk002 committed:
Update String#camelize to provide feedback when wrong option is passed

String#camelize was returning nil without any feedback when an
invalid option was passed as parameter. This update makes the method
to raises an ArgumentError when the option passed is invalid, similar
to what Ruby does for String#downcase (and others) in 2.4.1.

https://ruby-doc.org/core-2.4.1/String.html#method-i-downcase

guilherme committed:
Handling add/remove to/from migration edge cases

Making sure the table name is parsed correctly when an add/remove column migration have 'from'/'to' in the table name.

rafaelfranca committed:
Merge pull request #29984 from colorfulfool/form-helpers-unicode-sanitization

Allow non-English values for collection_radio_buttons

kamipo committed:
Passing `klass` to `StatementCache.new`

Actually `StatementCache#execute` is always passed the same klass that
the owner klass of the connection when the statement cache is created.
So passing `klass` to `StatementCache.new` will make more DRY.

y-yagi committed:
Remove unused `LoadError::REGEXPS` constant

since 4ad1a52, `LoadError::REGEXPS` is no longer needed.

Commit:
Fix ActiveStorage::Service::AzureService#url

Add in the content_type option, which is defined as part of the generic ActiveStorage::Service class.

Without this option, attempts to generate a URL for an attached file fail with "ArgumentError (unknown keyword: content_type)"

georgeclaghorn commented:
This PR is based on the active-storage-import branch, which isn't currently green.

dixpac committed:
Remove unecesarry exception variable

y-yagi committed:
Use `assert_nil` if `value` is nil in `assert_field_default_value`

It is deprecate to specify nil for expect argument of `assert_equal`.
Ref: https://github.com/seattlerb/minitest/commit/922bc9151a622cb3ef0b9f170aa09c3bb72c7eb8

sgrif committed:
Fix changelog format

sgrif committed:
Merge pull request #30039 from ricardotk002/fix-string-camelize-invalid-option

Update String#camelize to provide feedback when a wrong option is sent

sgrif committed:
Merge pull request #30011 from guilherme/fix-migration-generator-special-cases

Handling add/remove to/from migration edge cases

kamipo committed:
Add backticks around String#camelize

[ci skip]

sgrif committed:
Merge pull request #30050 from kamipo/dont_pass_connection_to_statement_cache_execute

Passing `klass` to `StatementCache.new`

gladtocode committed:
Revise bundle binstub error message

schneems committed:
Merge pull request #29967 from naw/binstub-error-msg

Improve bundler binstub error message

rafaelfranca committed:
Fix all rubocop violations

sgrif committed:
Merge pull request #30052 from y-yagi/deprecate_load_error_regexps

Deprecate `LoadError::REGEXPS` constant

kamipo committed:
Use `content_type.start_with?("...")` than `content_type =~ /^.../`

`start_with?` is a little faster than regexp for prefix matching by a
fixed string.

eileencodes committed:
Merge pull request #30022 from y-yagi/remove_extra_empty_line

Remove extra empty line

sgrif committed:
Merge pull request #30069 from kamipo/use_concurrent_map_than_mutex

Use `Concurrent::Map` than `Mutex` and `Mutex_m` for statement caches

rafaelfranca committed:
Merge pull request #30058 from y-yagi/use_assert_nil_if_value_is_nil_in_assert_field_default_value_

Use `assert_nil` if `value` is nil in `assert_field_default_value`

sgrif committed:
Merge pull request #29520 from kirs/serialize-vs-postgres-native-column

Do not let use `serialize` on native JSON/array column

Commit:
Merge pull request #30020 from rails/active-storage-import

Add Active Storage to Rails

dhh commented:
Can you rebase against master now that we've merged Active Storage? Thanks!

y-yagi committed:
Fix repository URL [ci skip]

changed `rails/activestorage` to `rails/rails`.

kamipo committed:
Merge pull request #30078 from y-yagi/fix_respository

Fix repository URL [ci skip]

claudiob committed:
`module ActiveStorage`, not `ActiveStorage::Class`

The reasons for this commit are:

- uniformity with the other Rails libraries
- (possibly) behave better with respect to autoloading
- fix the index in the generated documentation

Before this commit, run `rake rdoc` generates this left sidebar
(ActiveStorage entries are indexed twice, both inside and outside the module):

<img width="308" alt="before" src="https://user-images.githubusercontent.com/10076/28939523-7c087dec-7846-11e7-9289-38ed4a2930cd.png">

After this commit, run `rake rdoc` generates this left sidebar:
(ActiveStorage entries are only indexed inside the module):

<img width="303" alt="after" src="https://user-images.githubusercontent.com/10076/28939524-7c090be0-7846-11e7-8ee5-29dfecae548e.png">

georgeclaghorn committed:
Check for `app.secrets.secret_key_base`, not `app.config.secret_key_base`

By default, apps only have the former set.

y-yagi committed:
Fix ruby warnings

This fixes following warnings:

```
test/models/variant_test.rb:11: warning: ambiguous first argument; put parentheses or a space even after `/' operator
lib/active_storage/attached/macros.rb:63: warning: instance variable @active_storage_attached_highlights not initialized
lib/active_storage/attached/macros.rb:25: warning: instance variable @active_storage_attached_avatar not initialized
```

kamipo committed:
Add missing blank line between `config.active_storage` and `config.action_cable`

koic committed:
Change gem version of Active Storage to 5.2.0.alpha

matthewd committed:
Revert "Merge pull request #15446 from akshay-vishnoi/doc_changes"

It was right as originally written in #15440.

Commit:
Merge pull request #30056 from dixpac/remove_unecessary_variable

Remove unnecessary variable

Commit:
Merge pull request #30083 from koic/change_gem_version_of_active_storage_to_5_2_0

Change gem version of Active Storage to 5.2.0.alpha

Commit:
Merge pull request #30082 from kamipo/add_missing_blank_line

Add missing blank line between `config.active_storage` and `config.action_cable`

Commit:
Merge pull request #30080 from georgeclaghorn/active-storage-secret-key-base

Active Storage: check for `app.secrets.secret_key_base`, not `app.config.secret_key_base`

Commit:
Merge pull request #30081 from y-yagi/fix_ruby_warnings

Fix ruby warnings

Commit:
Merge pull request #30068 from kamipo/user_start_with_than_regexp

Use `content_type.start_with?("...")` than `content_type =~ /^.../`

Commit:
Merge branch 'master' into ast-module

Commit:
Merge pull request #30066 from claudiob/ast-module

`module ActiveStorage`, not `ActiveStorage::Class`

Commit:
Remove outdated comment [ci skip]

The comment was describing a previous version of the method with
a different signature. This is outdated since e76c38e

kamipo committed:
Merge pull request #30089 from yboulkaid/master

Remove outdated comment [ci skip]

vipulnsward committed:
Pass over Active Storage readme [ci skip] (#30086)

dixpac committed:
Fix wrong service name in the storage.yml

Commit:
Replaces the 17 for the actual call to params[:id] that would make the code work as intended. [ci skip]

kamipo committed:
Merge pull request #30097 from mgiagante/master

Replaces the 17 for the actual call to params[:id] that would make th…

bogdanvlviv committed:
Active Storage to 5.2.0.alpha inside `activestorage/package.json`

Follow #30083

Commit:
Merge pull request #30099 from bogdanvlviv/set_version_of_active_storage_to_5_2_0_alpha_inside_package_json

Active Storage to 5.2.0.alpha inside `activestorage/package.json`

Commit:
Merge pull request #30094 from dixpac/change_azure_name_in_storage_yml

Fix wrong service name in the storage.yml

maclover7 committed:
Lint actioncable/CHANGELOG.md

Postgres --> PostgreSQL
ActionCable --> Action Cable

[ci skip]

maclover7 committed:
Lint actionpack/CHANGELOG.md

Fixes a few grammar things.

[ci skip]

maclover7 committed:
Lint activerecord/CHANGELOG.md

Light grammar fixes, and added a few backticks.

[ci skip]

maclover7 committed:
Lint activesupport/CHANGELOG.md

Light grammar edits.

[ci skip]

maclover7 committed:
Lint railties/CHANGELOG.md

Light grammar edits.

[ci skip]

maclover7 committed:
Remove extra space between two sentences

[ci skip]

maclover7 committed:
Capitalize Rake

Should _probably_ be done across the board with all of our
documentation, but going to leave that for another time :)

[ci skip]

colorfulfool committed:
Extend image_tag to accept ActiveStorage Attachments and Variants (#30084)

* Extend image_tag to accept ActiveStorage's Attachments and Variants

* Flip resolve_image_source around

* Add tests for the new use-cases of image_tag

* Remove the higher-level test

* Update image_tag documentation

* Add error states into the test suite

* Re-raise polymorhic_url's NoMethodError as ArgumentError

* delegate_missing_to will raise DelegationError instead of NoMethodError

ffmike commented:
The combo of file renaming and other changes on master is beyond my git-fu. Given the small size of the actual code change, I'll just submit a new PR
